morebits.js

```diff
@@ -4001,6 +4001,71 @@ Morebits.wikitext.page.prototype = {
 		}
 	},
 
+	/**
+	 * Smartly insert a tag atop page text but after specified templates,
+	 * such as hatnotes, short description, or deletion and protection templates.
+	 * Notably, does *not* insert a newline after the tag
+	 *
+	 * @param {string} tag - The tag to be inserted
+	 * @param {string|array} regex - Templates after which to insert tag,
+	 * given as either as a string or an array to be joined by pipes
+	 * @param {string} [flags=i] - Regex flags to apply. Optional, defaults to /i
+	 * @param {string|array} preRegex - Optional string or array to match before
+	 * any template matches (i.e. before `{{`), such as html comments
+	 *
+	 * @returns {string}
+	 */
+	insertAfterTemplates: function(tag, regex, flags, preRegex) {
+		if (!tag) {
+			return this.text;
+		}
+
+		// .length is only a property of strings and arrays so we
+		// shouldn't need to check type
+		if (!regex || !regex.length) {
+			return tag + this.text;
```

Comment:
It's better to throw when the input is clearly invalid, so that the client can know they're doing something wrong.

Reply:
Fair, although I'm mixed on how much of an error/invalid we should actually consider these.  I guess it can just be handled on the "client" side, but this seemed nicer.  You're probably right about giving them the option of wanting to know.

Reply:
Went with both

Reply:
I suppose technically we could check the flags, something like `/^[gimsuy]$/`